backend/src/main/java/pt/ulisboa/tecnico/socialsoftware/tutor/course/CourseService.java

```diff
@@ -115,6 +123,24 @@ public void deactivateCourseExecution(int executionId) {
         courseExecution.setStatus(CourseExecution.Status.INACTIVE);
     }
 
+    @Retryable(
+            value = { SQLException.class },
+            backoff = @Backoff(delay = 5000))
+    @Transactional(isolation = Isolation.REPEATABLE_READ)
+    public void anonymizeCourseExecutionUsers(int executionId) {
+        CourseExecution courseExecution = courseExecutionRepository.findById(executionId).orElseThrow(() -> new TutorException(COURSE_EXECUTION_NOT_FOUND));
+        for (User user : courseExecution.getUsers()) {
+            String oldUsername = user.getUsername();
+            authUserRepository.delete(user.getAuthUser());
+            user.setAuthUser(null);
+            String newUsername = user.getUsername();
```

Comment:
Change the username in the db, do not redefine the getUsername, we want to anonymize even to the database admin

Reply:
Username is an AuthUser's property and without an object (`user.setAuthUser(null);` `orphanRemoval = true`), there is nothing to write.

Reply:
OK!